app/models/name.rb

```diff
@@ -408,33 +458,189 @@ class Name < AbstractModel
+          # Note small diff w :text_name_includes scope
+          where(Name[:text_name].matches("#{text_name} %"))
+        }
+  scope :subtaxa_of,
+        lambda { |name|
+          if name.at_or_below_genus?
+            # Subtaxa can be determined from the text_nam
+            subtaxa_of_genus_or_below(name.text_name).
+              with_correct_spelling
+          else
+            # Need to examine the classification strings
+            with_rank_and_name_in_classification(name.rank, name.text_name).
+              with_correct_spelling
+          end
+        }
 
-  scope :with_name_like,
-        ->(text_name) { where(Name[:text_name].matches("#{text_name} %")) }
+  ### Pattern Search
+  scope :include_synonyms_of,
+        lambda { |name|
+          where(id: name.synonyms.map(&:id)).with_correct_spelling
+        }
+  scope :include_subtaxa_of,
+        lambda { |name|
+          names = [name] + Name.subtaxa_of(name)
+          where(id: names.map(&:id)).with_correct_spelling
+        }
+  scope :text_name_includes,
+        ->(text_name) { where(Name[:text_name].matches("%#{text_name}%")) }
+  scope :with_classification, -> { where(Name[:classification].not_blank) }
+  scope :without_classification, -> { where(Name[:classification].blank) }
+  scope :classification_includes,
+        lambda { |classification|
+          where(Name[:classification].matches("%#{classification}%"))
+        }
+  scope :with_author, -> { where(Name[:author].not_blank) }
+  scope :without_author, -> { where(Name[:author].blank) }
+  scope :author_includes,
+        ->(author) { where(Name[:author].matches("%#{author}%")) }
+  scope :with_citation, -> { where(Name[:citation].not_blank) }
+  scope :without_citation, -> { where(Name[:citation].blank) }
+  scope :citation_includes,
+        ->(citation) { where(Name[:citation].matches("%#{citation}%")) }
+  scope :with_notes, -> { where(Name[:notes].not_blank) }
+  scope :without_notes, -> { where(Name[:notes].blank) }
+  scope :notes_include,
+        ->(notes) { where(Name[:notes].matches("%#{notes}%")) }
+  scope :with_comments, -> { joins(:comments).distinct }
+  scope :without_comments, -> { missing(:comments) }
+  scope :comments_include, lambda { |summary|
+    joins(:comments).where(Comment[:summary].matches("%#{summary}%")).distinct
+  }
+  scope :on_species_list,
+        lambda { |species_list|
+          joins(observations: :species_lists).
+            merge(SpeciesListObservation.where(species_list: species_list))
+        }
+  # Accepts region string, location_id, or Location instance
+  scope :at_location,
+        lambda { |location|
+          case location
+          when String # treat it as a region, not looking for all string matches
+            joins(observations: :location).
+              where(Location[:name].matches("%#{location}"))
+          when Integer, Location
+            joins(:observations).where(observations: { location: location })
+          end
+        }
```

Comment:
Consider revising this and Name.inbox to simply chain an Observation scope. Something like:
```suggestion
  scope :at_location,
        lambda { |location|
          joins(:observations).
          merge(Observation.at_location(location)
        }
```